src/pollers/mod.rs

```diff
@@ -132,6 +136,27 @@ pub trait ServerGatewayApis {
         result: Option<Payloads>,
     ) -> Result<RespondActivityTaskCompletedResponse>;
 
+    /// Report activity task heartbeat by sending details to the server. `task_token` contains activity
+    /// identifier that would've been received from [crate::Core::poll_activity_task] API.
+    /// `result` contains `cancel_requested` flag, which if set to true indicates that activity has been cancelled.
+    async fn record_activity_heartbeat(
+        &self,
+        task_token: Vec<u8>,
+        details: Option<Payloads>,
+    ) -> Result<RecordActivityTaskHeartbeatResponse>;
+
+    /// Report activity task heartbeat by sending details to the server. This function should only be used
+    /// when task token is not available, such as in case of manual completion, otherwise use `record_activity_heartbeat`.
+    /// `activity_id`, `workflow_id` and `run_id` must be set to respective IDs of activity, workflow and workflow run.
+    /// `result` contains `cancel_requested` flag, which if set to true indicates that activity has been cancelled.
+    async fn record_activity_heartbeat_by_id(
```

Comment:
Rather than making two separate APIs it may make sense to use an enum that is either a task token or this info

Reply:
That's what I've just ditched, I thought we were going to get rid of this one as well in favor of calling heartbeat by id from the lang?